cairo/ethereum/cancun/vm/instructions/block.cairo

```diff
@@ -287,6 +298,7 @@ namespace Internals {
         with stack {
             let err = push(U256(new U256Struct(block_hashes.value.low, block_hashes.value.high)));
             if (cast(err, felt) != 0) {
+                EvmImpl.set_stack{evm=evm}(stack);
```

Comment:
missing one before the ok below ?

Reply:
i think there's a bigger issue in this function, which is that it's taking both `evm` and a `stack` implicit argument.

so either stop takin the stack as implicit argument, and explicitly rebind the stack to the EVM, OR only take the stack as implicit argument and pass the other values explicitly.

Reply:
> i think there's a bigger issue in this function, which is that it's taking both evm and a stack implicit argument.

i dont think it is, this is a util that takes evm as explicit arg